File: csharp/DVSClient/DVSClient/server/phone/RestApiPhoneValidateRequest.cs
using Newtonsoft.Json;
using DVSClient.Phone;

namespace DVSClient.Server.Phone
{
    public class RestApiPhoneValidateRequest
    {
        [JsonProperty("number")]
        public string? Number { get; set; }

        [JsonProperty("output_format")]
        public string? OutputFormat { get; set; }

        [JsonProperty("cache_value_days")]
        public int? CacheValueDays { get; set; }

        [JsonProperty("country_iso")]
        public string? CountryIso { get; set; }

        [JsonProperty("get_ported_date")]
        public bool? GetPortedDate { get; set; }

        [JsonProperty("get_disposable_number")]
        public bool? GetDisposableNumber { get; set; }

        [JsonProperty("supplementary_live_status")]
        public RestApiPhoneSupplementaryLiveStatus? SupplementaryLiveStatus { get; set; }

        public static RestApiPhoneValidateRequest Using(Configuration phoneConfiguration)
        {
            var request = new RestApiPhoneValidateRequest();

            if (phoneConfiguration.OutputFormat != null)
            {
                request.OutputFormat = phoneConfiguration.OutputFormat;
            }

            if (phoneConfiguration.CacheValueDays.HasValue)
            {
                request.CacheValueDays = phoneConfiguration.CacheValueDays.Value;
            }

            if (phoneConfiguration.IncludePortedDate.HasValue)
            {
                request.GetPortedDate = phoneConfiguration.IncludePortedDate;
            }

            if (phoneConfiguration.IncludeDisposableNumber.HasValue)
            {
                request.GetDisposableNumber = phoneConfiguration.IncludeDisposableNumber;
            }

            if (phoneConfiguration.LiveStatusForLandline != null || phoneConfiguration.LiveStatusForMobile != null)
            {
                var supplementaryLiveStatus = new RestApiPhoneSupplementaryLiveStatus();

                if (phoneConfiguration.LiveStatusForMobile != null)
                {
                    supplementaryLiveStatus.Mobile = phoneConfiguration.LiveStatusForMobile.ToList().ConvertAll(c => c.Iso3Code);
                }

                if (phoneConfiguration.LiveStatusForLandline != null)
                {
                    supplementaryLiveStatus.Landline = phoneConfiguration.LiveStatusForLandline.ToList().ConvertAll(c => c.Iso3Code);
                }

                request.SupplementaryLiveStatus = supplementaryLiveStatus;
            }

            return request;
        }
    }
}

File: typescript/src/server/phone/restApiPhoneValidateRequest.ts
import { PhoneConfiguration } from "../../phone/phoneConfiguration";
import { RestApiPhoneSupplementaryLiveStatus } from "./restApiPhoneSupplementaryLiveStatus";

export type RestApiPhoneValidateRequest = {
    number?: string;
    output_format?: string;
    cache_value_days?: number;
    country_iso?: string;
    get_ported_date?: boolean;
    get_disposable_number?: boolean;
    supplementary_live_status?: RestApiPhoneSupplementaryLiveStatus;
};

export function getPhoneValidateRequestFromConfig(config: PhoneConfiguration): RestApiPhoneValidateRequest {
    const request: RestApiPhoneValidateRequest = {};
    if (config.options.outputFormat) {
        request.output_format = config.options.outputFormat;
    }
    if (config.options.cacheValueDays) {
        request.cache_value_days = config.options.cacheValueDays;
    }
    if (config.options.includePortedDate) {
        request.get_ported_date = config.options.includePortedDate;
    }
    if (config.options.includeDisposableNumber) {
        request.get_disposable_number = config.options.includeDisposableNumber;
    }
    if (config.options.liveStatusForMobile || config.options.liveStatusForLandline) {
        const liveStatus: RestApiPhoneSupplementaryLiveStatus = {};
        if (config.options.liveStatusForMobile) {
            liveStatus.mobile = config.options.liveStatusForMobile.map(country => country.iso3Code);
        }
        if (config.options.liveStatusForLandline) {
            liveStatus.landline = config.options.liveStatusForLandline.map(country => country.iso3Code);
        }
        request.supplementary_live_status = liveStatus;        

    }
    return request;
}